master/buildbot/steps/trigger.py

```diff
@@ -80,33 +80,28 @@ def end(self, result):
             return self.finished(result)
 
     # Create the properties that are used for the trigger
-    def createTriggerProperties(self):
+    def createTriggerProperties(self, properties):
         # make a new properties object from a dict rendered by the old
         # properties object
         trigger_properties = Properties()
-        trigger_properties.update(self.set_properties, "Trigger")
+        trigger_properties.update(properties, "Trigger")
         return trigger_properties
 
-    # Get all scheduler instances that were configured
-    # A tuple of (triggerables, invalidnames) is returned
-    def getSchedulers(self):
-        all_schedulers = self.build.builder.botmaster.parent.allSchedulers()
-        all_schedulers = dict([(sch.name, sch) for sch in all_schedulers])
-        invalid_schedulers = []
-        triggered_schedulers = []
-        # don't fire any schedulers if we discover an unknown one
-        for scheduler in self.schedulerNames:
-            scheduler = scheduler
-            if scheduler in all_schedulers:
-                sch = all_schedulers[scheduler]
-                if ITriggerableScheduler.providedBy(sch):
-                    triggered_schedulers.append(sch)
-                else:
-                    invalid_schedulers.append(scheduler)
-            else:
-                invalid_schedulers.append(scheduler)
-
-        return (triggered_schedulers, invalid_schedulers)
+    def getSchedulerByName(self, name):
+        # Use a quick cache to avoid generating this dict every time.
+        try:
+            all_schedulers = self._all_schedulers
+        except AttributeError:
```

Comment:
using exceptions as a flow control?

why not set `self._all_schedulers` to `None` and check against that?